tests/filtered_relation/tests.py

```diff
@@ -0,0 +1,351 @@
+            condition=Q(book__title__iexact='poem by alice')
+        ).extra(where=["1 = 1"]).order_by(
+            F('book_alice__id').desc(nulls_last=True)),
+            ["<Author: Alice>", "<Author: Jane>"])
+
+    @skipUnlessDBFeature('supports_select_union')
+    def test_filtered_relation_union(self):
+        qs1 = Author.objects.filtered_relation(
+            'book', alias='book_alice',
+            condition=Q(book__title__iexact='poem by alice')
+        ).filter(book_alice__isnull=False).order_by()
+        qs2 = Author.objects.filtered_relation(
+            'book', alias='book_jane',
+            condition=Q(book__title__iexact='the book by jane a')
+        ).filter(book_jane__isnull=False).order_by()
+        self.assertQuerysetEqual(qs1.union(qs2).order_by('id'), ["<Author: Alice>", "<Author: Jane>"])
+
+    @skipUnlessDBFeature('supports_select_intersection')
+    def test_filtered_relation_intersection(self):
+        qs1 = Author.objects.filtered_relation(
+            'book', alias='book_alice',
+            condition=Q(book__title__iexact='poem by alice')
+        ).filter(book_alice__isnull=False).order_by()
+        qs2 = Author.objects.filtered_relation(
+            'book', alias='book_jane',
+            condition=Q(book__title__iexact='the book by jane a')
+        ).filter(book_jane__isnull=False).order_by()
+        self.assertQuerysetEqual(qs1.intersection(qs2).order_by('id'), [])
+
+    @skipUnlessDBFeature('supports_select_difference')
+    def test_filtered_relation_difference(self):
+        qs1 = Author.objects.filtered_relation(
+            'book', alias='book_alice',
+            condition=Q(book__title__iexact='poem by alice')
+        ).filter(book_alice__isnull=False).order_by()
+        qs2 = Author.objects.filtered_relation(
+            'book', alias='book_jane',
+            condition=Q(book__title__iexact='the book by jane a')
+        ).filter(book_jane__isnull=False).order_by()
+        self.assertQuerysetEqual(qs1.difference(qs2).order_by('id'), ["<Author: Alice>"])
+
+    def test_filtered_relation_select_for_update(self):
+        self.assertQuerysetEqual(Author.objects.filtered_relation(
+            'book', alias='book_jane',
+            condition=Q(book__title__iexact='the book by jane a')
+        ).filter(book_jane__isnull=False).select_for_update(),
+            ['<Author: Jane>'])
+
+    def test_filtered_relation_defer(self):
+        # One query for the list, and one query to fetch the
+        # deferred title.
+        with self.assertNumQueries(2):
+            self.assertQuerysetEqual(Author.objects.filtered_relation(
+                'book', alias='book_alice',
+                condition=Q(book__title__iexact='poem by alice')
+            ).select_related('book_alice').defer('book_alice__title').order_by(
+                F('book_alice__title').desc(nulls_last=True)),
+                ["Poem by Alice", None], lambda author: getattr(author.book_alice, 'title', None))
```

Comment:
`None` in `getattr` is already default.

Reply:
If you don't specify the default you end up with `AttributeError`.